Web.Client/Pages/Electives/StudentSubjectRegistrationComponent.razor.cs

```diff
@@ -1,39 +1,59 @@
 ﻿using Havit;
 using MensaGymnazium.IntranetGen3.Contracts;
 using MensaGymnazium.IntranetGen3.Primitives;
+using MensaGymnazium.IntranetGen3.Web.Client.Services.DataStores;
 
 namespace MensaGymnazium.IntranetGen3.Web.Client.Pages.Electives;
 
 public partial class StudentSubjectRegistrationComponent
 {
-	[Parameter] public int? SubjectId { get; set; }
+	[Parameter] public int? SubjectId { get; set; } // Xopa: why is this nullable?
 
 	[Parameter] public EventCallback OnRegistrationChanged { get; set; }
 
 	[Inject] protected IHxMessageBoxService MessageBox { get; set; }
 	[Inject] protected IHxMessengerService Messenger { get; set; }
 	[Inject] protected ISubjectRegistrationsManagerFacade SubjectRegistrationsManagerFacade { get; set; }
+	[Inject] protected IStudentSubjectRegistrationsDataStore StudentSubjectRegistrationsDataStore { get; set; }
```

Comment:
S tímhle jediným si nejsem jist, jestli je dobrý nápad cachovat registrace na klientovi v DataStore. Přijde mi, že to je transakčních charakter dat, který by se měl vždy brát čerstvý ze serveru (API).